js/components/graph/Sidebar.js

```diff
@@ -0,0 +1,182 @@
+import Focus from "./Focus";
+
+export default class Sidebar extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      contentHidden: true,
+      focusDisabled: false,
+      focusActive: false,
+      graphActive: false,
+      graphName: "",
+      toggled: false
+    }
+  }
+
+  componentWillUpdate(prevProps) {
+    // Check to see if we have a graph from local storage on load
+    if (this.state.graphName.length == 0 && prevProps.graphName !== this.state.graphName) {
+      this.setState({ graphName: prevProps.graphName }, () => {
+        this.handleGraph();
+      });
+    }
+  }
+
+  handleGraph = graphName => {
+    if (graphName != undefined) {
+      this.props.getGraph(graphName);
+    } else {
+      graphName = this.state.graphName;
+    }
+    // Enable Focuses nav if CS graph is selected
+    if (graphName === "Computer Science") {
+      this.setState({
+        focusDisabled: false,
+      });
+    } else {
+      this.setState({
+        focusDisabled: true,
+      });
+    }
+  }
+
+  createGraphButtons = () => {
+    const graphButtons = [];
+    this.props.graphs.forEach((graph, i) => {
+      const graphId = "graph-" + graph.id;
+      const graphName = graph.title;
+      graphButtons.push(
+        <div
+          className="graph-button"
+          id={graphId}
+          key={i}
+          onClick={() => this.handleGraph(graphName)}
+        >
+          {graphName}
+        </div>
+      );
+    });
+    return graphButtons;
+  }
+
+  getFocusData = () => {
+    const computerScienceFocusData = [
+      ["sci", "Scientific Computing"],
+      ["AI", "Artificial Intelligence"],
+      ["NLP", "Natural Language Processing"],
+      ["vision", "Computer Vision"],
+      ["systems", "Computer Systems"],
+      ["game", "Video Games"],
+      ["HCI", "Human Computer Interaction"],
+      ["theory", "Theory of Computation"],
+      ["web", "Web Technologies"],
+    ];
+    const focusComponents = [];
+    computerScienceFocusData.forEach((focus, i) => {
+      const openDetails = this.props.currFocus == focus[0];
+      focusComponents.push(
+        <Focus
+          key={i}
+          pId={focus[0]}
+          focusName={focus[1]}
+          openDetails={openDetails}
+          highlightFocus={(id) => this.props.highlightFocus(id)}
+        />
+      )
+    })
+    return focusComponents;
+  }
+
+  toggleSidebar = location => {
+    if (this.state.toggled) {
+      // close graph
+      this.setState({
+        contentHidden: true,
+        focusActive: false,
+        graphActive: true,
+        toggled: false,
+      })
+    } else if (!this.state.toggled && location === "button") {
+      // open graph
+      this.setState({
+        toggled: true,
+        contentHidden: false,
+        graphActive: true,
+      });
+    }
+  }
+
+  showFocuses = focus => {
+    if (focus) {
+      // show focuses
+      this.setState({
+        focusActive: true,
```

Comment:
It's redundant to have two separate state values that are negations of each other. Instead, have a single state that is an index (either 0 or 1) of which "tab" is open. This will also allow us to generalize to multiple tabs in the future, e.g. for the dynamic graph form.